lib/src/components/NxToast/NxToast.tsx

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { useState, useContext, useEffect } from 'react';
+import classnames from 'classnames';
+import FocusContext from './contexts';
+
+import { nxToastPropTypes, NxToastProps, ToastModel } from './types';
+
+export { ToastModel };
+
+const NxToast = (props: NxToastProps) => {
+  const { onClose, children } = props;
+
+  const [isClosing, setIsClosing] = useState(false);
+
+  const focusedToast = useContext(FocusContext);
+
+  useEffect(()=>{
+    focusedToast?.adjustFocus();
+    return () => { Promise.resolve().then(() => focusedToast?.adjustFocus()); };
+  }, []);
+
+  const handleClose = () => {
+    if (!isClosing) {
+      setIsClosing(true);
+    }
+  };
+
+  const handleAnimationEnd = () => {
+    if (isClosing) {
+      onClose();
+    }
+  };
+
+  const validChild = React.Children.only(children);
+  const childrenWithProps = React.isValidElement(children) ?
+    React.cloneElement(validChild, { onClose: handleClose }) : validChild;
```

Comment:
If it's not a valid element, I don't think we want to silently continue, we want to throw an exception

Reply:
> If it's not a valid element, I don't think we want to silently continue, we want to throw an exception

a throw new TypeError added here: [b666e9f](https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/b666e9f46cff2e89cbbdd062e9c3e79ef7d8bbd7)